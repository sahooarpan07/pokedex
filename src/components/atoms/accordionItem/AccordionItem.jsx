import React, {useState} from 'react';
import {View, Text, Pressable} from 'react-native';
import {styles} from './AccordionItem.styles';

const AccordionItem = ({title, children}) => {
  const [expanded, setExpanded] = useState(false);

  return (
    <View style={styles.container}>
      <View style={styles.header}>
        <Text style={styles.title}>{title}</Text>
        <Text style={styles.status}>Status</Text>
        <Pressable
          testID="expand-button"
          style={styles.expandedBtn}
          onPress={() => setExpanded(!expanded)}>
          <Text style={styles.expandedBtnText}>+</Text>
        </Pressable>
      </View>
      {expanded && (
        <View style={styles.body} testID="accordion-children">
          {children}
        </View>
      )}
    </View>
  );
};

export default AccordionItem;
