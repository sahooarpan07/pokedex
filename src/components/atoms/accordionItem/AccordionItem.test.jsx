/**
 * @format
 */

import 'react-native';
import React from 'react';
import AccordionItem from './AccordionItem';
import {Text} from 'react-native';
import {render, fireEvent, screen} from '@testing-library/react-native';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('should render Accordion Item', () => {
  const tree = renderer
    .create(
      <AccordionItem title="Type">
        <Text> Testing Accordion</Text>
      </AccordionItem>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

test('should display children when expanded', () => {
  render(
    <AccordionItem title="Type">
      <Text> Testing Accordion</Text>
    </AccordionItem>,
  );
  const accordionIcon = screen.getByTestId('expand-button');
  fireEvent.press(accordionIcon);
  const accordionChild = screen.getByTestId('accordion-children');
  expect(accordionChild).toBeDefined();
});
