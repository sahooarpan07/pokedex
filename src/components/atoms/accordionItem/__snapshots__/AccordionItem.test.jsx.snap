// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Accordion Item 1`] = `
<View
  style={
    {
      "borderColor": "#180345",
      "borderRadius": 5,
      "borderWidth": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "paddingHorizontal": 20,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      style={
        {
          "borderRightColor": "#ddd",
          "borderRightWidth": 1,
          "color": "#180345",
          "flex": 1,
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    >
      Type
    </Text>
    <Text
      style={
        {
          "color": "#180345",
          "flex": 2,
          "fontSize": 18,
          "paddingLeft": 10,
        }
      }
    >
      Status
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "borderColor": "#180345",
          "borderRadius": 30,
          "borderWidth": 1,
          "justifyContent": "center",
          "paddingHorizontal": 8,
          "paddingVertical": 2,
        }
      }
      testID="expand-button"
    >
      <Text
        style={
          {
            "color": "#180345",
            "fontSize": 18,
          }
        }
      >
        +
      </Text>
    </View>
  </View>
</View>
`;
